### ISSUE TITLE
Blog post topics as part of documentation

### ISSUE THREAD

**Issue by mitjade** [sysgears/apollo-universal-starter-kit#393]
I would like to start a discussion of the blog post titles, that would add most value to better understanding of this kit, and then we can work together through PRs and write them when we have some spare time.

My recommendations:
1. Kit philosophy
Describe what is the purpose of this kit. What it tries to solve. What all it incorporates and makes it work out of the box. Showing of backend, fronted, mobile elements and all tools that ware build to help development easier like spin, logs, eslint, prettier, flow, cli, etc.

2. Module system
Explaining the philosophy and why this is is useful. Showing of how to add and remove them through `cli` and basic step by step walk through how to add a bd table, creating a schema entry, making a resolver and then displaying this data on web and mobile.

3. Example modules
Explain each example module in more detail, especially user module.

**Comment by mitjade** [comment]
If this could be done with a video tutorial, this would also be great, especially a walk through of all the features, but I do not feel comfortable speaking, since my english is not all that good :)

**Comment by vlasenko** [comment]
Good topic, thank you Mitja!

I would like to add the following blow posts:
1. Testing Apollo GraphQL React container components. 
Describe our approach to testing UI components, what we can test, how we mock GraphQL schema and Apollo Link. 
2. Testing Apollo GraphQL API.
How we setup testing of GraphQL queries, mutations and subscriptions.
3. Webpack Hot Module Replacement for each target platform of the kit for hot code reload.
Here we can write 3 articles - one for the server, one for the web and one for React Native.

**Comment by vlasenko** [comment]
4. Universal components that work on all platforms - write once, run everywhere.
Describe our approach to universal components and solutions of common problems.

**Comment by mitjade** [comment]
@vlasenko I would feel comfortable in trying to explain the `Module system` with some of your input and the two module examples I wrote (`post` and `user`).

The rest I think only you are really able to explain in detail how this works, but I can help out with some parts.

Will let you know if and when I will start on this, so we are no writing the same things.

**Comment by mitjade** [comment]
@vlasenko @mairh I think if we dedicate just an hour or so a week to this we could be able to make great progress over a couple of months in explaining this parts.

**Comment by vlasenko** [comment]
@mitjade Sure Mitja, sounds good!

**Comment by mitjade** [comment]
@vlasenko I also had to do some filtering and sorting on one of my projects and would like to share my approach, but is quite hard to incorporate this into the post example, since it is a bit complex because of pagination and adding new elements without refreshing the whole list.

Should this be demonstrated in a separate example? I tried to follow the approach of https://www.graph.cool/docs/reference/simple-api/ordering-by-field-vequoog7hu/ and https://www.graph.cool/docs/reference/simple-api/filtering-by-field-xookaexai0/ without automatically generating any filter parameters, just the naming conventions.

**Comment by vlasenko** [comment]
@mitjade Sure, another example is fine

**Comment by mairh** [comment]
Yeah, filtering and sorting would be a nice addition.

Regarding, the blog post, I think we have noted down most of the points over here, but it might prove to be difficult to organize them. We could start slow: for example, start with the general idea behind the kit and have the blog post in different parts/series.

We can choose to post the stuff on Medium and at the end of every article we can say something like "Stay tuned for the next post explaining blah blah"

**Comment by mitjade** [comment]
@mairh Yes. We need to break this up into small meaningful parts that can be written quickly and not just started and never finished.

**Comment by mitjade** [comment]
Maybe we could pick one small part and all work on it for a week through a PR and then publish it, decide on the next one and so on...

**Comment by vlasenko** [comment]
@mitjade Yep, interesting idea

**Comment by mairh** [comment]
So the blog is gonna be on Medium?

**Comment by vlasenko** [comment]
@mairh I guess yes, do you have other ideas about the place for the blog?

**Comment by mairh** [comment]
No, I think that's one of the best places to get the audience.

**Comment by mairh** [comment]
Do you have a publication created already? If you have a publication then other people can also write blog articles and publish under your publication.

**Comment by vlasenko** [comment]
@mairh I have no publications on medium yet.

**Comment by mairh** [comment]
You can create one using the following: https://help.medium.com/hc/en-us/articles/115004681607-Create-publication

Perhaps, think of a good name for it.
